File: qdm-tob/qdm-tob/tob-backend/tob-app/src/main/java/cn/qdm/tob/modules/product/catalog/domain/ProductCatalog.java
package cn.qdm.tob.modules.product.catalog.domain;

import cn.qdm.tob.modules.product.catalog.enums.ProductStatus;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 商品资料实体（主键 = 销售大区 + 商品条码）
 */
@Data
@TableName("prd_product_catalog")
public class ProductCatalog {

    @TableId
    private Long id;

    /** 销售大区编号 */
    private String salesRegionCode;

    /** 销售大区名称（冗余） */
    private String salesRegionName;

    /** 商品条码 */
    private String productBarcode;

    /** 商品名称（条码反查） */
    private String productName;

    /** 仓库编码（SAP） */
    private String warehouseCode;

    /** 仓库名称（冗余） */
    private String warehouseName;

    /** 状态：LISTED=已上架 UNLISTED=已下架 */
    private ProductStatus status;

    /** 小程序名称（商品别名） */
    private String miniappName;

    /** 商品主图 URL */
    private String mainImage;

    /** 轮播图 URL 列表（JSON 数组，最多 9 张） */
    private String carouselImages;

    /** 商品详情（富文本） */
    private String productDetail;

    /** 订购基数 */
    private BigDecimal orderBaseQty;

    /** 订购下限 */
    private BigDecimal orderMinQty;

    /** 订购上限 */
    private BigDecimal orderMaxQty;

    /** 每日可用库存 */
    private BigDecimal dailyStock;

    /** 今日可用数量（=库存-已售） */
    private BigDecimal dailyAvailable;

    /** 今日已售数量（订单同步） */
    private BigDecimal dailySold;

    /** 创建人 */
    private String createdBy;

    /** 修改人 */
    private String updatedBy;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 修改时间 */
    private LocalDateTime updatedAt;
}

File: qdm-tob/qdm-tob/tob-frontend/tob-frontend-web/src/api/modules/product-catalog.ts
import request from '@/utils/request'

// ================================================================
// 类型定义
// ================================================================

export interface ProductCatalogViewVO {
  id: number
  salesRegionCode: string
  salesRegionName: string
  productBarcode: string
  productName: string
  warehouseCode: string
  warehouseName: string
  status: string
  miniappName: string
  mainImage: string
  carouselImages: string
  productDetail: string
  orderBaseQty: number
  orderMinQty: number
  orderMaxQty: number
  dailyStock: number
  dailyAvailable: number
  dailySold: number
  createdBy: string
  updatedBy: string
  createdAt: string
  updatedAt: string
}

export interface ProductCatalogCreationVO {
  salesRegionCode: string
  salesRegionName: string
  productBarcode: string
  warehouseCode: string
  warehouseName: string
  status?: string
  miniappName: string
  mainImage: string
  carouselImages: string
  productDetail: string
  orderBaseQty: number
  orderMinQty: number
  orderMaxQty: number
  dailyStock: number
  createdBy: string
}

export interface ProductCatalogEditVO {
  warehouseCode: string
  warehouseName: string
  status: string
  miniappName: string
  mainImage: string
  carouselImages: string
  productDetail: string
  orderBaseQty: number
  orderMinQty: number
  orderMaxQty: number
  updatedBy: string
}

export interface StockAdjustVO {
  newDailyStock: number
}

export interface SkuLookupVO {
  barcode: string
  productName: string
  spec: string
  unit: string
}

export interface WarehouseSimpleVO {
  code: string
  name: string
  region: string
}

export interface ImportResultVO {
  createdCount: number
  updatedCount: number
  failedCount: number
  errors: ImportError[]
}

export interface ImportError {
  row: number
  reason: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

export interface ApiResponse<T> {
  code: number
  data: T
  msg: string
}

export interface ProductCatalogQuery {
  pageNum?: number
  pageSize?: number
  salesRegionCode?: string
  productBarcode?: string
  productName?: string
  status?: string
}

// ================================================================
// API 方法
// ================================================================

/** 分页查询 */
export function getProductCatalogPage(params: ProductCatalogQuery) {
  return request.get<unknown, ApiResponse<PageResult<ProductCatalogViewVO>>>(
    '/api/admin/product/catalog/list',
    { params },
  )
}

/** 详情 */
export function getProductCatalogDetail(id: number) {
  return request.get<unknown, ApiResponse<ProductCatalogViewVO>>(
    '/api/admin/product/catalog/detail',
    { params: { id } },
  )
}

/** 新增 */
export function createProductCatalog(data: ProductCatalogCreationVO) {
  return request.post<unknown, ApiResponse<null>>('/api/admin/product/catalog', data)
}

/** 编辑 */
export function updateProductCatalog(id: number, data: ProductCatalogEditVO) {
  return request.put<unknown, ApiResponse<null>>('/api/admin/product/catalog', data, {
    params: { id },
  })
}

/** 库存调整 */
export function adjustStock(id: number, data: StockAdjustVO) {
  return request.put<unknown, ApiResponse<null>>('/api/admin/product/catalog/adjust-stock', data, {
    params: { id },
  })
}

/** 条码反查 */
export function lookupBarcode(barcode: string) {
  return request.get<unknown, ApiResponse<SkuLookupVO>>(
    '/api/admin/product/catalog/lookup-barcode',
    { params: { barcode } },
  )
}

/** 仓库列表 */
export function getWarehouses(salesRegionCode?: string) {
  return request.get<unknown, ApiResponse<WarehouseSimpleVO[]>>(
    '/api/admin/product/catalog/warehouses',
    { params: { salesRegionCode } },
  )
}

/** 导出 */
export function exportProductCatalog(params: ProductCatalogQuery) {
  return request.get('/api/admin/product/catalog/export', {
    params,
    responseType: 'blob',
    headers: { Action: 'export' },
  })
}

/** 导入 */
export function importProductCatalog(file: File) {
  const formData = new FormData()
  formData.append('file', file)
  return request.post<unknown, ApiResponse<ImportResultVO>>(
    '/api/admin/product/catalog/import',
    formData,
    { headers: { 'Content-Type': 'multipart/form-data' } },
  )
}
